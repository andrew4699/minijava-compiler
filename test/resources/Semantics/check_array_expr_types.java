class a {

  public static void main(String[] args) {
    System.out.println(1);
  }
}

class b {
  public boolean valid() {
    int[] x; int b;
    b = 10;
    x = new int[b];
    x[1] = 0;
    x[b - 5] = 1;
    x[4] = x[b - 5];
    x[x.length - 1] = x.length;

    return true;
  }

  public boolean invalid() {
    int[] x; int b;

    x = new int[false];
    x[x] = 2;
    x[1] = true;
    b = 0;
    x[2] = b.length;

    return false;
  }
}