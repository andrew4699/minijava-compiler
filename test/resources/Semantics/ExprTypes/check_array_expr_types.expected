[ExprTypesVisitor - 24]: Expression yielded type boolean where int was expected.
[ExprTypesVisitor - 25]: Expression yielded type int[] where int was expected.
[ExprTypesVisitor - 26]: Expression yielded type boolean where int was expected.
[ExprTypesVisitor - 28]: Expression yielded type int where int[] was expected.
